src/scripts/import-needs-assessment-data/add-surveys.ts

```diff
@@ -0,0 +1,288 @@
+  });
+
+  console.log("Adding items completed!");
+
+  const validSurveys: Survey[] = [
+    ...resultsMap[UploadWorkflowStatus.SUCCESS],
+    ...resultsMap[UploadWorkflowStatus.ALREADY_EXISTS],
+  ].reduce((surveys: Survey[], workflow: SurveyUploadWorkflow) => {
+    return [...surveys, workflow.data];
+  }, [] as Survey[]);
+
+  return validSurveys;
+}
+
+const isFulfilled = <T>(
+  value: PromiseSettledResult<T>,
+): value is PromiseFulfilledResult<T> => {
+  return value.status === "fulfilled";
+};
+
+const _isRejected = <T>(
+  value: PromiseSettledResult<T>,
+): value is PromiseRejectedResult => {
+  return value.status === "rejected";
+};
+
+/*  Consolidate Surveys
+ * ------------------------------------------------------- */
+export function consolidateSurveys(data: NeedAssessment[]): string[] {
+  const uniqueSurveys = new Set<string>();
+
+  data.forEach((item) => {
+    const surveyString = `${item.survey.year}-${item.survey.quarter}`;
+    uniqueSurveys.add(surveyString);
+  });
+
+  return Array.from(uniqueSurveys);
+}
+
+/*  Parse Surveys
+ * ------------------------------------------------------- */
+function parseSurvey({
+  data,
+  orig,
+  status,
+  logs,
+}: SurveyUploadWorkflow): SurveyUploadWorkflow {
+  logs = [...logs, `Log: parsing survey "${orig}"`];
+
+  if (typeof orig === "string") {
+    //if orig is a string, validate it
+    const [year, quarter] = orig.split("-");
+    if (!year || !quarter) {
+      throw {
+        data,
+        orig,
+        status: UploadWorkflowStatus.ORIGINAL_DATA_INVALID,
+        logs: [
+          ...logs,
+          `Error: Invalid survey input: "${orig}". Expected a non-null string in YYYY-QQ format.`,
+        ],
+      };
+    }
+  } else if (typeof orig === "object" && orig !== null) {
+    // if orig is an object, validate it
+    if (!orig.year || !orig.quarter) {
+      throw {
+        data,
+        orig,
+        status: UploadWorkflowStatus.ORIGINAL_DATA_INVALID,
+        logs: [
+          ...logs,
+          `Error: Invalid survey input: "${JSON.stringify(orig)}". Expected a non-null object with year and quarter.`,
+        ],
+      };
+    }
+  } else {
+    throw {
+      data,
+      orig,
+      status: UploadWorkflowStatus.ORIGINAL_DATA_INVALID,
+      logs: [
+        ...logs,
+        `Error: Invalid survey input: "${orig}". Expected a non-null value.`,
+      ],
+    };
+  }
+
+  const newData = {
+    ...data,
+    survey: orig,
+  };
+
+  return {
+    data: newData,
+    orig,
+    status,
+    logs,
+  };
+}
+
+/*  Get Surveys
+ * ------------------------------------------------------- */
+export async function getSurvey({
+  data,
+  orig = `${data.year}-${data.quarter}`,
+  status,
+  logs,
+}: SurveyUploadWorkflow): Promise<SurveyUploadWorkflow> {
+  logs = [
+    ...logs,
+    `Log: Checking if NeedsAssessment.Survey "${orig}" already exists.`,
+  ];
+
+  //Fetch the data from Strapi
+  const response = await fetch(`${STRAPI_ENV.URL}/surveys?`, {
```

Comment:
I think this is fine, I am assuming we might want to filter surveys by year, quarter, or a specific need.